x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/SearchHitRowSet.java

```diff
@@ -84,14 +83,10 @@
         indexPerLevel = new int[maxDepth + 1];
         this.innerHit = innerHit;
 
-        TotalHits totalHits = response.getHits().getTotalHits();
-
         // compute remaining limit (only if the limit is specified - that is, positive).
         int remaining = limit < 0 ? limit : limit - size;
         // either the search returned fewer records than requested or the limit is exhausted
-        if (size < sizeRequested || remaining == 0
-        // or exactly `totalHits` records have been fetched
-            || totalHits != null && totalHits.value == hits.length) {
```

Comment:
Good point, the only case I know about is `COUNT(*)` on implicit aggregations though (and in this case size is always 1). Are there others? Since it's not tested I'd personally prefer to keep it out for now. I think it's also not quite correct as is because it does not consider the relation (eq vs gte).